### ISSUE TITLE
Testers needed!

### ISSUE THREAD

**Issue by XePeleato** [XePeleato/android_device_huawei_p8lite_alel21#5]
Alright,  I have finally built TWRP 3.0 so I need some people to test it.

Just drop a comment if you'd like to provide feedback!

![recovery](https://cloud.githubusercontent.com/assets/6912584/13899735/8d721d0e-edf5-11e5-97d3-524e3e85fdcb.jpg)

**Comment by FelipeBR28** [comment]
I can't test it yet (because it's being repaired)  but when it's finished, I'd like to test it.

**Comment by olrak2** [comment]
Very good news, how did you solve the black screen?

I'm very excited to test it :)

**Comment by XePeleato** [comment]
Well, I had to edit a graphics file.

http://error404software.com/android/recovery.zip

Unzip and run "fastboot flash recovery recovery.img"

**Comment by olrak2** [comment]
Installation without problem :)

All settings are working for the moment.  

Will test other settings in the couple of days.

Very good job.  Keep up the good work

**Comment by Mast3rPK** [comment]
Hii, TWRP is perfect :dancer: 

![TWRP 3 Test](http://imageshack.com/a/img922/8775/v8uk6R.jpg)

**Comment by FelipeBR28** [comment]
And, @XePeleato , can it be flashed by Flashify? It's easier 😅

**Comment by XePeleato** [comment]
@Tuentista At the moment, it can't, since I just released it here to get a few people to provide some feedback, so far I am pretty happy with it, so I might release a .zip

@Mast3rPK Thank you for your feedback! I really appreciate it.

**Comment by olrak2** [comment]
If you can't flash it with fastboot you are not worth of being a tester :stuck_out_tongue_winking_eye: 

@XePeleato hope development on your AOSP rom is going well

**Comment by FelipeBR28** [comment]
Obviously I can, just said that it's easier, not else.

**Comment by debbiche** [comment]
@XePeleato is there a source for this somewhere? Would love to try to build it for the P8 ;)

**Comment by XePeleato** [comment]
@debbiche I am talking with the members of TeamWin, so they can provide me with an 'official' recovery tree, the problem was that since the graphics are now coded in C++ instead of C, the  graphics.c patch by jniklast became obsolete and I had to adapt it. Whenever that change is pushed you'll be able to build it.

**Comment by bboci80** [comment]
help me please!
possible to, installing the Android6.0. marshmallow  on the huawei P8 Lite Dual SIM(ALE-L21)? 
if the answeris yes, then how i beginning andme it;;m?

**Comment by XePeleato** [comment]
@bboci80 There are some posts on XDA about it, however I won't update until Huawei releases it officially.

**Comment by muhammadfahadbaig** [comment]
@XePeleato Recovery works great :+1:

**Comment by muhammadfahadbaig** [comment]
@XePeleato I can't seem to reboot to system from TWRP. It just powers off the device instead of rebooting.

**Comment by XePeleato** [comment]
@muhammadfahadbaig Have you played with the kernel? Did that happen with 2.8.7 version?

**Comment by muhammadfahadbaig** [comment]
It's the same with 2.8.7.0. (I built both of them myself using your device tree for Android 5.1.1 version with my own kernel built from source). I want to stay away from kernel sources as far as I can. I'll now use the Image from stock boot.img...

**Comment by XePeleato** [comment]
Have you tried with the file I released above?

**Comment by muhammadfahadbaig** [comment]
Same with your recovery.img...

**Comment by olrak2** [comment]
@XePeleato , probably not your recovery's fault but wanted to share my experience.  

I had a backup of my rom: version B188 in TWRP (external SD Card)

I installed version B052 (did a full wipe, all also internal memory etc.)  and then with OTA to B170.

Installed your recovery TWRP to go back to my backup of the B188.

Once my backup did install i booted up to my backup, no problem but my rom asked to make an OTA update to B188.

So there must be a partition or a file that did not got backup by TWRP.  Maybe cust.img or something else but i think that can cause for problem when you restore a stock rom.

**Comment by muhammadfahadbaig** [comment]
@olrak2 You're right about the partition. Cust.img holds all current firmware information. That's why if you want to downgrade from lets say Android 6.0 to Android 5.1.1 you need to flash cust.img of Android 5.1.1 and then the whole stock ROM. It would be easy to add Cust partition to the Backup menu. Let me see it and I'll try to build one with this option.

**Comment by muhammadfahadbaig** [comment]
@olrak2 Just add this line in recovery.fstab (either before compiling from source or decompiling the recovery.img and editing this file in /ramdisk/etc/).
`/cust      ext4    /dev/block/platform/hi_mci.0/by-name/cust flags=backup=1;display="Cust";wipeingui`

**Comment by muhammadfahadbaig** [comment]
@XePeleato As for the black screen patch for HiSilicon SoC,
does FBIOPUT_VSCREENINFO always passes or fails for our chipset?

**Comment by XePeleato** [comment]
I've never tried it , the custom graphics file works fine for me
